platform/security/common/src/main/java/org/codice/ddf/security/common/Security.java

```diff
@@ -62,12 +66,15 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(Security.class);
 
-    private static final String INSUFFICIENT_PERMISSIONS_ERROR = "Current user doesn't have sufficient privileges to run this command";
+    private static final String INSUFFICIENT_PERMISSIONS_ERROR =
+            "Current user doesn't have sufficient privileges to run this command";
 
     private static final RolePrincipal ADMIN_ROLE = new RolePrincipal("admin");
 
     private Subject cachedSystemSubject;
 
+    private static final Map<String, Subject> CACHED_GUEST_SUBJECT = new ConcurrentHashMap<>();
```

Comment:
What will prevent this from growing unbounded? Better to use a Guava cache perhaps, with the guts of your `getGuestSubject` method in a loader method.